### ISSUE TITLE
Suggest: LDAP auth for WebDAV

### ISSUE THREAD

**Issue by basisau** [SeedDMS/webdav_checkout#1]
Related topic, based on research: https://sourceforge.net/p/seeddms/tickets/250/

Currently running SeedDMS 6.0.24, PHP 8.1.16, in a Docker container on Debian 11.
Have installed and configured webdav_checkout 1.1.2.

SeedDMS LDAP integration to AD is working, users can sign in to SeedDMS web app using AD creds.

Attempting to access \\<serverfqnd>@SSL\checkout from Windows Explorer results in Access Denied (0x80040005) for LDAP-based users, with a log entry of "[error] 'webdav_checkout': wrong password"

More specifically, attempting to access WebDAV from Windows Explorer appears to be possible only if:
1) The LDAP user identity has already signed into the web app at least once. It's assumed this is required because it creates a local user record in the local database.
2) The user's password has also been manually set e.g. by an Administrator via Admin Tools > User Management > (Choose user) > (Type the user's password twice) > (click Save)

Once steps (1) and (2) above are completed, the user an access WebDAV and the system works fine.

The suggestion is to check LDAP for user auth, not just local user database, within webdav_checkout.

**Comment by SeedDMS** [comment]
You are right. The extension currently doesn't use the internal authentication method. I need to fix that.

**Comment by basisau** [comment]
Please let us know if donating would assist with this feature, we're interested in contributing and it's a great product, including this extension.

We may also be able to help with coding if you happen to have an OVA-based dev environment? I'm sure the code can draw on the patterns already established for the LDAP query done for web app sign-in. We haven't had time to set up a build environment so a virtualised dev VM would be an ideal starting point.

**Comment by SeedDMS** [comment]
I updated the code to use the generally used authentication method. That makes it authenticate against an ldap server but it does not create a user in the database if it doesn't exist. Donation do always help :-)
